### PR TITLE
RI Workshop Improvements

### PR DESCRIPTION
**Pull request by chrisking**
*Issue #, if available:*

*Description of changes:*

1. Full support for continual interactions and to render them to a user.
1. Creates the bucket in notebook
1. Adds batch recommendations
1. Does not filter ratings.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: getting_started/RI_PersonalizeWorkshop.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    Properties:
+      InstanceType: "ml.t2.medium"
+      NotebookInstanceName: !Ref NotebookName
+      RoleArn: !GetAtt SageMakerIamRole.Arn
+      VolumeSizeInGB: !Ref VolumeSize
+      DefaultCodeRepository: https://github.com/aws-samples/amazon-personalize-samples.git
+
+Outputs:
+  S3Bucket:
+    Value: !Ref BucketName
```
Comment (dotgc) [inline getting_started/RI_PersonalizeWorkshop.yaml]: We should remove this too

Comment (chrisking) [reply getting_started/RI_PersonalizeWorkshop.yaml]: Please check now.

File: getting_started/ReInvent2019_Workshop.ipynb [hunk 2]
```diff
@@ -941,29 +974,170 @@
...
+   "outputs": [],
+   "source": [
+    "s3 = boto3.client('s3')\n",
+    "export_name = json_input_filename + \".out\"\n",
+    "s3.download_file(bucket_name, export_name, export_name)\n",
     "\n",
-    "recommendations_df = recommendations_df.join(new_rec_DF)\n",
+    "# Update DF rendering\n",
+    "pd.set_option('display.max_rows', 30)\n",
+    "with open(export_name) as json_file:\n",
```
Comment (dotgc) [inline getting_started/ReInvent2019_Workshop.ipynb]:
```
results = []
with open(export_name) as json_file:
    for row in json_file:
        line = json.loads(row)
        user_id = line['input']['userId']
        # Create a list for all the movies
        recommendation_list = []
        # Add all the entries
        for item in line['output']['recommendedItems']:
            title = get_movie_title(item)
            recommendation_list.append(title)
        results.append({'userId': user_id, 'recommendations': recommendation_list})

recommendations_df = pd.DataFrame(results, columns=('userId', 'recommendations'))

Comment (chrisking) [reply getting_started/ReInvent2019_Workshop.ipynb]: The way this is framed it renders the User ID as the header and the recommendations just follow. Yours would have a lot of duplicate user_ID values and would require a pivot to render clearly to users.

File: getting_started/ReInvent2019_Workshop.ipynb [hunk 3]
```diff
@@ -151,7 +148,11 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "bucket = \"personalizereinventchrisking\" # Replace the values within the quote marks with your bucket name"
+    "s3 = boto3.client('s3')\n",
+    "account_id = boto3.client('sts').get_caller_identity().get('Account')\n",
+    "bucket_name = account_id + \"reinventforecastworkshop\"\n",
```
[the remark this answers is not in the document]
Comment (chrisking) [reply getting_started/ReInvent2019_Workshop.ipynb]: oof, fixing now.  Its rough covering both services all day.

[the remark this answers is not in the document]
Comment (chrisking) [reply getting_started/ReInvent2019_Workshop.ipynb]: Fixed in 8ad79de90b212faa72700c6a2bdf85c188c1befb

**Pull request merged**